File: 프로그래머스/unrated/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.js
function solution(n) {
  const sequence = [];

  while (n >= 1) {
    
    sequence.push(n);

    
    if (n === 1) {
      break;
    }

    
    if (n % 2 === 0) {
      n /= 2;
    } else {
      n = n * 3 + 1;
    }
  }

  return sequence;
}

File: 프로그래머스/unrated/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(int n) {
        List<Integer> array = new ArrayList<>();
        
        while(n != 1){
            array.add(n);
            if(n % 2 == 0){
                n/=2;
            }else{
                n = 3 * n + 1;
            }   
        }
        array.add(n);
        
        int result[] = new int [array.size()];
        for(int i = 0; i < array.size(); i++){
            result[i] = array.get(i);
        }
        return result;
    }
}